src/libAtomVM/nifs.gperf

```diff
@@ -120,6 +120,7 @@ atomvm:add_avm_pack_binary/2, &atomvm_add_avm_pack_binary_nif
 atomvm:add_avm_pack_file/2, &atomvm_add_avm_pack_file_nif
 atomvm:close_avm_pack/2, &atomvm_close_avm_pack_nif
 atomvm:read_priv/2, &atomvm_read_priv_nif
+atomvm:set_time_of_day/2, &atomvm_set_time_of_day_nif
```

Comment:
I like this new function since it is very useful, however I'm worried that adding it to atomvm module might turn it to a kind of kitchen sink.
discuss: should we create a dedicated module for cross-platform functions aimed at interacting with the underlying operating system? or should we place it anyway in some other place?

Reply:
I had a version of this branch where the Nif was in the ESP32 platform only, since it doesn’t build on stm32 and its utility on UNIX is questionable.  Happy to revert it back to that, if that addresses your concern.

Reply:
We already have several atomvm:posix_* functions, we call move them all to a dedicated posix module.

Reply:
Changed to `atomvm:posiz_clock_settime/2`